332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeTaskUtils.java

```
@@ -615,16 +615,20 @@ public Map<String, String> removeEnvVarsWithSecretRef(Map<String, String> envVar
 
   public void checkSecretAccess(Ambiance ambiance, List<SecretVariableDetails> secretVariableDetails,
       String accountIdentifier, String projectIdentifier, String orgIdentifier) {
-    List<EntityDetail> entityDetails =
-        secretVariableDetails.stream()
-            .map(secretVariableDetail -> {
-              return createEntityDetails(secretVariableDetail.getSecretVariableDTO().getSecret().getIdentifier(),
-                  accountIdentifier, projectIdentifier, orgIdentifier);
-            })
-            .collect(Collectors.toList());
-
-    if (isNotEmpty(entityDetails) && featureFlagService.isEnabled(CIE_ENABLED_RBAC, accountIdentifier)) {
-      pipelineRbacHelper.checkRuntimePermissions(ambiance, entityDetails, false);
+    try {
+      List<EntityDetail> entityDetails =
+          secretVariableDetails.stream()
+              .map(secretVariableDetail -> {
+                return createEntityDetails(secretVariableDetail.getSecretVariableDTO().getSecret().getIdentifier(),
+                    accountIdentifier, projectIdentifier, orgIdentifier);
+              })
+              .collect(Collectors.toList());
+
+      if (isNotEmpty(entityDetails) && featureFlagService.isEnabled(CIE_ENABLED_RBAC, accountIdentifier)) {
+        pipelineRbacHelper.checkRuntimePermissions(ambiance, entityDetails, false);
+      }
+    } catch (NullPointerException npe) {
```

Can we catch a generic exception here and instead of just logging throw an error with a generic message ?